src/mnemonics/electrum-words.cpp

```diff
@@ -431,39 +438,41 @@ namespace crypto
       }
       const std::vector<std::string> &word_list = language->get_word_list();
       // To store the words for random access to add the checksum word later.
-      std::vector<std::string> words_store;
+      std::vector<epee::wipeable_string> words_store;
 
       uint32_t word_list_length = word_list.size();
       // 4 bytes -> 3 words.  8 digits base 16 -> 3 digits base 1626
-      for (unsigned int i=0; i < len/4; i++, words += ' ')
+      for (unsigned int i=0; i < len/4; i++, words.push_back(' '))
       {
-        uint32_t w1, w2, w3;
-        
-        uint32_t val;
+        uint32_t w[4];
 
-        memcpy(&val, src + (i * 4), 4);
+        memcpy(&w[0], src + (i * 4), 4);
 
-        w1 = val % word_list_length;
-        w2 = ((val / word_list_length) + w1) % word_list_length;
-        w3 = (((val / word_list_length) / word_list_length) + w2) % word_list_length;
+        w[1] = w[0] % word_list_length;
+        w[2] = ((w[0] / word_list_length) + w[1]) % word_list_length;
+        w[3] = (((w[0] / word_list_length) / word_list_length) + w[2]) % word_list_length;
 
-        words += word_list[w1];
+        words += word_list[w[1]];
         words += ' ';
-        words += word_list[w2];
+        words += word_list[w[2]];
         words += ' ';
-        words += word_list[w3];
+        words += word_list[w[3]];
+
+        words_store.push_back(word_list[w[1]]);
+        words_store.push_back(word_list[w[2]]);
+        words_store.push_back(word_list[w[3]]);
 
-        words_store.push_back(word_list[w1]);
-        words_store.push_back(word_list[w2]);
-        words_store.push_back(word_list[w3]);
+        memwipe(w, sizeof(w));
       }
 
-      words.pop_back();
-      words += (' ' + words_store[create_checksum_index(words_store, language->get_unique_prefix_length())]);
+      char c;
+      do c = words.pop_back(); while(c != ' ');
+      words += ' ';
+      words += words_store[create_checksum_index(words_store, language->get_unique_prefix_length())];
```

Comment:
Doesn't this line alone suffice (without `pop_back` and `+= ' '`) since the above for loop does `, words.push_back(' ')` at the end of each iteration?

Reply:
I don't follow. The loop does not push a ' ' at the end AFAICT.

Reply:
The loop does push `' '` at L448:
```c++
      for (unsigned int i=0; i < len/4; i++, words.push_back(' '))
```

Reply:
Whee. I'd missed it. So I need an extra pop I think. I'll see what I need to repro/test it.

Reply:
There was some confusion as to what the code was trying to do. It was adding a space after removing it, and I'd missed the add in the for statement. I've simplified now.